src/ipahealthcheck/meta/core.py

```diff
@@ -25,31 +26,30 @@ class MetaCheck(Plugin):
     def check(self):
 
         rval = constants.SUCCESS
-        if not os.path.exists(paths.FIPS_MODE_SETUP):
-            fips = "missing {}".format(paths.FIPS_MODE_SETUP)
-            logger.debug('%s is not installed, skipping',
-                         paths.FIPS_MODE_SETUP)
+        if not os.path.exists(paths.PROC_FIPS_ENABLED):
+            fips = "missing {}".format(paths.PROC_FIPS_ENABLED)
+            logger.debug("Can't find %s, skipping" %
+                         paths.PROC_FIPS_ENABLED)
         else:
             try:
-                result = ipautil.run([paths.FIPS_MODE_SETUP,
-                                      '--is-enabled'],
-                                     capture_output=True,
-                                     raiseonerr=False,)
+                proc_fips_enable_path = Path(paths.PROC_FIPS_ENABLED)
+                result_text = proc_fips_enable_path.read_text()
+                result = int(result_text)
```

Comment:
There is a platforms call in freeipa that would do much of this for you, is_fips_enabled(). Now in that we consider FIPS not enabled if the file doesn't exist so we could alternatively take that stance as well. It would be consistent anyway.
I added the timeout part in case executing the command took a while. It wasn't exactly fast. I think that part can be dropped. And the freeipa version of this just looks for value != 0. So I think this can be simplified by using the same call as freeipa itself does.

Reply:
Then again, the freeipa test is a hammer while the healthcheck test is more nuanced. I guess I could go either way.

Reply:
You're right, there's the `is_fips_enabled` in the `BaseTaskNamespace` class, it would be probably cleaner that way.

Reply:
Ultimately I've decided to keep it this way, just got rid of the timeout. Even though most of the errors are very unlikely, we should provide more user feedback than just *enabled*/*disabled* if something goes really wrong.